toolbox

```diff
@@ -477,6 +478,105 @@ list_containers()
 )
 
 
+run_rm()
+(
+    force_remove=$1
+    command_to_run=$2
+    id_or_name=$3
+    keyword=$4
+
+    if $force_remove; then
+        if ! rm_output=$($prefix_sudo podman $command_to_run $id_or_name --force 2>&42); then
+            echo "$0: failed to remove $id_or_name"
+            exit 1
+        fi
+    else
+        if ! rm_output=$($prefix_sudo podman $command_to_run $id_or_name 2>&42); then
+            echo "$0: failed to remove $id_or_name"
+            exit 1
+        fi
+    fi
+    if ! [ "$rm_output" = "" ]; then
+        echo -e "${LRC}$keyword $rm_output removed${NC}"
+    fi
+)
+
+
+remove_image()
+(
+    id_or_name_list=$1
+    force_remove=$2
+    remove_all=$3
+
+    if ! $remove_all; then
+        IFS=' '
+        export IFS
+        for id_or_name in $id_or_name_list; do
+            if ! inspect_out=$($prefix_sudo podman inspect $id_or_name --type image --format \"{{.Labels}}\" 2>&42); then
+                echo "$0: failed to inspect $id_or_name"
+                exit 1
+            fi
+            if ! [[ $inspect_out = *"com.redhat.component:fedora-toolbox"* ]]; then
+                echo "Not a fedora-toolbox image"
+                exit 1
+            fi
+            run_rm "$force_remove" "rmi" "$id_or_name" "image"
+        done
+    else
+        if ! output=$($prefix_sudo podman images \
+                --filter "label=com.redhat.component=fedora-toolbox" \
+                --format "{{.ID}}" 2>&42); then
+            echo "$0: failed to list all images"
+            exit 1
+        fi
+        if [ "$output" = "" ]; then
+            echo "No images available"
+            exit
+        fi
+        while read -r line_id; do
+            run_rm "$force_remove" "rmi" "$line_id" "image"
+        done <<< "$output"
+    fi
+)
+
+remove_container()
+(
+    id_or_name_list=$1
+    force_remove=$2
+    remove_all=$3
+
+    if ! $remove_all; then
+        IFS=' '
+        export IFS
+        for id_or_name in $id_or_name_list; do
+            if ! inspect_out=$($prefix_sudo podman inspect $id_or_name --type container --format \"{{.Config.Labels}}\" 2>&42); then
+                echo "$0: failed to inspect $id_or_name"
+                exit 1
+            fi
+            if ! [[ $inspect_out = *"com.redhat.component:fedora-toolbox"* ]]; then
+                echo "Not a fedora-toolbox container"
+                exit 1
+            fi
+            run_rm "$force_remove" "rm" "$id_or_name" "container"
+        done
+    else
+        if ! output=$($prefix_sudo podman ps --all \
+                --filter "label=com.redhat.component=fedora-toolbox" \
+                --format "{{.ID}}" 2>&42); then
+            echo "$0: failed to list all containers"
```

Comment:
Using the entire `$0` seems too noisy. We should use just the `basename`, like we do elsewhere.

Reply:
Done.